src/ls/driver.ts

```diff
@@ -111,20 +113,20 @@ export default class ExasolDriver extends AbstractDriver<DriverLib, DriverOption
     const db = await this.open();
     const splitQueries = parse(queries);
 
-    const responseData: QueryResponse = await this.queue.add(
-      () => new Promise(
-        (resolve, reject) => db.com({ 'command': 'executeBatch', 'sqlTexts': splitQueries },
-          resolve,
-          this.rejectErr(reject))
-      )
-    );
-    const res = [];
-    for (let index = 0; index < responseData.results.length; index++) {
-      const result = responseData.results[index];
+    const responses: QueryResponse[] = await Promise.all<QueryResponse>(
+      splitQueries.map((query) => this.queue.add(() =>
```

Comment:
this means all the queries will run one after the other, as the queue allows only one query at a time, rather than having all queries run at once (Although, come to think of it, I assume that Exasol would run them sequentially anyway, as there might be dependencies between them,)

Reply:
I think this is a good thing, as we don't want to run many queries simultaneously. I am only worried that first all queries are run and only after that the results are fetched so all resultsets are open when the fetches start and only begin closing one by one.
But fixing that is a another task IMO.